lib/android.js

```diff
@@ -31,19 +32,32 @@ class JavaOnPathCheck extends DoctorCheck {
 
 // Check tools
 class AndroidToolCheck extends DoctorCheck {
-  constructor (toolName, toolPath) {
+  constructor () {
     super();
-    this.toolName = toolName;
-    this.toolPath = toolPath;
+    this.tools = ['adb', 'android', 'emulator'];
   }
 
   async diagnose () {
     if (typeof process.env.ANDROID_HOME === 'undefined') {
-      return nok(`${this.toolName} could not be found because ANDROID_HOME is NOT set!`);
+      return nok(`${this.tools.join(', ')} could not be found because ANDROID_HOME is NOT set!`);
     }
-    let fullPath = path.resolve(process.env.ANDROID_HOME, this.toolPath);
-    return await fs.exists(fullPath) ? ok(`${this.toolName} exists at: ${fullPath}`) :
-      nok(`${this.toolName} could NOT be found at '${fullPath}'!`);
+
+    // TODO: Prevent printing logs
+    const adb = await ADB.createADB();
+    const no = [];
+    for (const binary of this.tools) {
+      try {
+          await adb.getBinaryFromSdkRoot(binary);
```

Comment:
I'd rather modify appium-adb to export this method statically it does not depend on adb instance itself. This will also resolve the issue with logging

Reply:
Yeah, i think so, too